core/include/gnuradio-4.0/Port.hpp

```diff
@@ -868,9 +868,9 @@ samples_to_next_tag(const PortLike auto &port) {
     }
 
     // at least one tag is present -> if tag is not on the first tag position read up to the tag position
-    const auto &tagData           = port.tagReader().get();
```

Comment:
This return by reference is important because the port takes care of the lifetime and keeps a reader copy. Dropping the reference here invokes the copy constructor which will create and register a new `CircularBuffer::buffer_reader`

Notably see [here](https://github.com/fair-acc/graph-prototype/blob/caccc351279d55a9f825830f25d79ada6badf676/core/include/gnuradio-4.0/CircularBuffer.hpp#L482-L486):
```cpp
        buffer_reader(std::shared_ptr<buffer_impl> buffer) noexcept :
            _buffer(buffer), _size(buffer->_size) {
            gr::detail::addSequences(_buffer->_read_indices, _buffer->_cursor, {_read_index});
            _read_index_cached = _read_index->value();
        }
```